src/esp/agent/Agent.cpp

```diff
@@ -31,10 +32,21 @@ Agent::Agent(scene::SceneNode& agentNode, const AgentConfiguration& cfg)
     // sensor
 
     auto& sensorNode = agentNode.createChild();
-    sensors_.add(
-        sensor::PinholeCamera::create(sensorNode, spec));  // transformed within
+    switch (spec->sensorSubtype) {
+      case sensor::SensorSubtype::PINHOLE: {
+        sensors_.add(sensor::PinholeCamera::create(sensorNode, spec));
+        break;
+      }
+      case sensor::SensorSubtype::ORHTOGRAPHIC: {
+        sensors_.add(sensor::OrthoCamera::create(sensorNode, spec));
+        break;
+      }
+      default: {
+        sensors_.add(sensor::PinholeCamera::create(sensorNode, spec));
```

Comment:
Shouldn't we error?

Reply:
I'm defaulting to pinhole because that field originally (sensorSubType) was ignored. If you think an error would be better choice for this, I can do that too.

Reply:
I am sayiing we add one more check if it doesn't equal "pinhole," then we complain. I'd rather not have it fail silently.